app/src/main/java/woowacourse/paint/PaintBinding.kt

```diff
@@ -0,0 +1,37 @@
+package woowacourse.paint
+
+import androidx.databinding.BindingAdapter
+import com.google.android.material.slider.RangeSlider
+
+@BindingAdapter("app:setColorPalette")
+fun BoardView.setColorPalette(colorPalette: ColorPalette?) {
+    if (colorPalette == null) return
+    changeColor(colorPalette)
+}
+
+@BindingAdapter("app:setWidth")
+fun BoardView.setWidth(width: Float?) {
+    if (width == null) return
+    setWidth(width)
+}
+
+@BindingAdapter("app:eraseAll")
+fun BoardView.eraseAll(isErasing: Boolean?) {
+    if (isErasing == null) return
+    if (isErasing) erase()
+}
+
+@BindingAdapter("app:setupSlider")
+fun RangeSlider.setupSlider(viewModel: MainViewModel) {
+    this.valueFrom = MIN_WIDTH_VALUE
+    this.valueTo = MAX_WIDTH_VALUE
+
+    this.addOnChangeListener(
+        RangeSlider.OnChangeListener { _, value, _ ->
+            viewModel.changeWidth(value)
+        },
+    )
+}
```

Comment:
바인딩 어댑터는 전역에서 접근할 수 있는만큼 신중하게 작성해야 한다고 생각합니다.
MainViewModel을 받는 것보다 함수를 받도록 해서 하면 더 좋을 거 같아요!

아 그리고 리스너 등록은 저도 배운건데 아래와 같은 방법으로도 작성 가능합니다!
```suggestion
this.addOnChangeListener { _, value, _ ->
    viewModel.changeWidth(value)
}
```

Reply:
저도 써니의 의견에 아주아주 동의합니다! 리뷰주신 것처럼 함수를 받도록 수정하려고 하다가 bindingAdapter로 인자가 필요한 함수를 파라미터로 받을 수 없다는 사실을 알게되었습니다. 굉장한 삽질하다가 interface로 해결했습니다👍[참고 링크](https://stackoverflow.com/questions/55578601/typed-parameter-in-a-lambda-not-recognized-by-the-data-binder-when-using-binding)